src/processTargets/processSelectionType.ts

```diff
@@ -173,6 +175,78 @@ function processParagraph(
   };
 }
 
+function processNonWhitespaceSequence(
+  target: PrimitiveTarget,
+  selection: SelectionWithEditor,
+  selectionContext: SelectionContext
+) {
+  const { selectionType, insideOutsideType, position, modifier } = target;
+  const { document } = selection.editor;
+  const { start, end } = selection.selection;
+  let newSelection;
+
+  let startLine;
+  let i = start.line;
+  do {
+    startLine = document.lineAt(i++);
+  } while (startLine.isEmptyOrWhitespace && i <= end.line);
```

Comment:
Why can't you just use `start.line` directly?

Reply:
Because that one might be empty. This solves the corner case where one of the edges are in an empty part of the document

Reply:
If it's empty, then the start is not part of a non-white space sequence so that should be an error

Reply:
I've implemented this port
```
Move it forward until the end of the next non-whitespace block
Move it backward to the end of the preceding non-whitespace block
```
I think that makes total sense and the implementation as it is right now is really useful in my opinion

Reply:
Yes but we discussed yesterday and decided to throw an error if an endpoint is not touching a non-whitespace sequence

Reply:
Also I don't think "take paint file" will work today, will it? Can't specify two selection types

Reply:
And "take paint file" violates the semantics of saying a scope type, which implies smallest containing instance of that scope type, not largest contained instance

Reply:
That's misunderstanding I said that we should throw an error if it was in complete white space. that the ends are is fine I think.

It will work with new grammar. It's already implemented and is useful for the future.

Reply:
We already have sub word modifier. I see no reason why modify can't both grow and shrink. 
A is now paint selects all content the current selection touches.

Reply:
So would you argue that if your selection is larger than a function, it should shrink to the function? What happens if it is larger than one function but contained in another? Just seems a bit ambiguous to allow "funk" to both shrink and expand. To me the semantics of `containingScope` are that it expands

Reply:
But I haven't implemented this as a containing scope. It is a selection type.

Containing scopes by their name is a bit different. But definitely to select all functions that your selection touches would probably be useful.

Reply:
But I'm not married to the solution so if you hate it in can change you to something simpler

Reply:
In the future there will be no distinction between containing scope and selection type so I'd just treat this as if it were selection type

I guess I'm not opposed to changing the semantics of containing scope to just be every instance of that scope that is touched by the input, just want to make sure we're consistent

I think that's actually pretty close to what we do with "token", right?

Reply:
I know but in this future we don't have to stick to the name containing scope if we don't want to. 

Yes we token we are growing but we don't shrink. But the question is if we shouldn't treat all the scope modifiers like this instead? I think we have to have a proper discussion about it

Reply:
To give some context the reason I think this fits particularly well with paint is that we already allow start and end to be in different sequences. Something we don't do containing scopes. If we already decided that this one is going to be treated differently I think allowing it to shrink is totally fine even if we don't implement it on the others scopes. I think we ide can have it as wildcard with its own rules or it should adhere to the same as containing scopes where both ends of the selection has to be inside and an error is thrown if it's not.

Reply:
We could def discuss, but fwiw I'm happy to just agree to change the semantics of today's `containingScope` modifier to be "any instance of the scope type touching the input to the modifier", which would necessitate your impl here, and then we could just merge.  That way we don't need to block on having a discussion.  But if you wanna talk it through before we merge I'm of course happy to do that

The one catch is that we'll then want to change impl of "round", "funk", etc to be consistent with these semantics.  Probably not urgent because it's kind of a corner case but just a thought

We could just merge this one, then file an issue to make sure to discuss / think through the semantics of containing scope (or whatever we rename it to to reflect that it's really "touching scope")

Reply:
I want to have a quick discussion about it before. I'm available before today's meeting if you are?

Reply:
As discussed, let's expand start cursor to containing sequence and end cursor to containing sequence, but never shrink, just throwing an error of either end is not contained in a non-whitespace sequence

Reply:
done